backend/codesirius/problems/views/language.py

```
@@ -0,0 +1,149 @@
+            logger.info(f"Language with ID: {pk} fetched successfully")
+        except Language.DoesNotExist:
+            logger.warning(f"Language with ID: {pk} not found")
+            raise NotFound(f"Language with ID: {pk} not found")
+        serializer = LanguageSerializer(instance=language)
+        return CodesiriusAPIResponse(data=serializer.data)
+
+    def put(self, request: Request, pk: int) -> CodesiriusAPIResponse:
+        """
+        Update a language by its primary key
+        """
+        logger.info(f"Updating language with ID: {pk}")
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)